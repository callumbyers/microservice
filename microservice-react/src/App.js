import React, { useState } from 'react';
import download from 'downloadjs';
import './App.css';
import { Card, Table } from './elements';
import { Input, CircularProgress } from '@material-ui/core';

function App() {
  const [ uploadedFiles, setUploadedFiles] = useState([]);
  const [ loading, setLoading ] = useState(false);

  const downloadDocument = (link) => {
    download(link);
  };

  const uploadDocument = (evt) => {
    const fileToUpload = evt.target.files[0];

    if(fileToUpload) {
      fetch('/file', {
        method: 'POST',
        body: JSON.stringify({
          fileName: fileToUpload.name
      })
      }).then((r) => r.json()).then((data) => {
        let formToSend = new FormData();

        for(var k in data.s3Post.fields) {
          formToSend.append(k, data.s3Post.fields[k]);
        }

        formToSend.append('file', fileToUpload, fileToUpload.name);

        try {
          setLoading(true);
          fetch(data.s3Post.url, {
            method: 'POST',
            body: formToSend
          }).then(() => {
            fetch(`fileUrl`, {
              method: 'POST',
              body: JSON.stringify({
                Key: data.s3Post.fields.key
              })
            }).then((t) => t.json())
              .then((p) => {
              const newState = [...uploadedFiles, {
                name: fileToUpload.name,
                url: p.url  
              }]

              setUploadedFiles(newState);
              setLoading(false);
            })
          })
      }

      catch(error) {
        console.log('Error: ', error);
        setLoading(false);
      }
    })
  }
}

  return (
    <div className="App">
      <div className='App-header'>
        <Card>
            <Input
              type='file'
              id='TestInput'
              disableUnderline
              fullWidth
              onChange={uploadDocument}
            />
            {loading ?
              <CircularProgress />
              :
            <Table
              data={uploadedFiles}
              onClick={downloadDocument}
            />
            }
        </Card>
      </div>
    </div>
  );
}

export default App;
