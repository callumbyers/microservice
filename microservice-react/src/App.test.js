import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

test("renders correctly", () => {
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
});
