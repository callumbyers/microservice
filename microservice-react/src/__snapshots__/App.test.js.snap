// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <div
      className="MuiPaper-root MuiCard-root Card-default-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input MuiInput-input"
          id="TestInput"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="file"
        />
      </div>
      <div
        className="MuiTableContainer-root"
      >
        <table
          aria-label="simple table"
          className="MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Download
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
    </div>
  </div>
</div>
`;
